### PR TITLE
adding Kwai-Klear/Klear-46B-A2.5B-Instruct

### PR DESCRIPTION
**Pull request by Goekdeniz-Guelmez**

### CODE REVIEW

**Comment by Goekdeniz-Guelmez** [comment]
python -m mlx_lm.generate --model /Volumes/T7_Shield/MODELS/MLX/Klear-46B-A2.5B-Instruct-3bit --prompt "Who is Michael Jackson"
<frozen runpy>:128: RuntimeWarning: 'mlx_lm.generate' found in sys.modules after import of package 'mlx_lm', but prior to execution of 'mlx_lm.generate'; this may result in unpredictable behaviour
Calling `python -m mlx_lm.generate...` directly is deprecated. Use `mlx_lm.generate...` or `python -m mlx_lm generate ...` instead.
==========
Michael Jackson (1958–2009) was an American singer, songwriter, and performer widely regarded as one of the most influential cultural icons in modern history. Dubbed the "King of Pop," he revolutionized music and pop culture with his groundbreaking artistry, distinctive voice, and pioneering music videos.

**Key Facts about Michael Jackson:**

- **Career:** Began his career as the lead singer of The Jackson 5 at age 11, later pursuing a solo
==========
Prompt: 30 tokens, 45.282 tokens-per-sec
Generation: 100 tokens, 71.893 tokens-per-sec
Peak memory: 20.246 GB

**Comment by Goekdeniz-Guelmez** [comment]
python -m mlx_lm.lora \
--train \
--num-layers 1 \
--model /Volumes/T7_Shield/MODELS/MLX/Klear-46B-A2.5B-Instruct-3bit \
--data mlx-community/wikisql \
--adapter-path "/Users/gokdenizgulmez/Desktop/adapters" \
--batch-size 1 \
--val-batches 1 \
--iters 10 \
--learning-rate 1e-5 \
--steps-per-report 1 \
--steps-per-eval 5 \
--max-seq-length 12 \
--grad-checkpoint
Calling `python -m mlx_lm.lora...` directly is deprecated. Use `mlx_lm.lora...` or `python -m mlx_lm lora ...` instead.
Loading pretrained model
Loading datasets
Loading Hugging Face dataset mlx-community/wikisql.
Training
Trainable parameters: 0.000% (0.070M/46214.683M)
Starting training..., iters: 10
Calculating loss...: 100%|███████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  5.73it/s]
Iter 1: Val loss 3.659, Val took 0.177s
Iter 1: Train loss 3.756, Learning Rate 1.000e-05, It/sec 0.553, Tokens/sec 6.080, Trained Tokens 11, Peak mem 20.236 GB
[WARNING] Some sequences are longer than 12 tokens. The longest sentence 80 will be truncated to 12. Consider pre-splitting your data to save memory.
Iter 2: Train loss 3.659, Learning Rate 1.000e-05, It/sec 5.822, Tokens/sec 64.047, Trained Tokens 22, Peak mem 20.246 GB
[WARNING] Some sequences are longer than 12 tokens. The longest sentence 103 will be truncated to 12. Consider pre-splitting your data to save memory.
Iter 3: Train loss 3.812, Learning Rate 1.000e-05, It/sec 9.428, Tokens/sec 103.710, Trained Tokens 33, Peak mem 20.246 GB
[WARNING] Some sequences are longer than 12 tokens. The longest sentence 66 will be truncated to 12. Consider pre-splitting your data to save memory.
Iter 4: Train loss 3.886, Learning Rate 1.000e-05, It/sec 9.394, Tokens/sec 103.339, Trained Tokens 44, Peak mem 20.246 
Calculating loss...: 100%|███████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 10.83it/s]
Iter 5: Val loss 3.693, Val took 0.095s
Iter 5: Train loss 3.881, Learning Rate 1.000e-05, It/sec 9.460, Tokens/sec 104.059, Trained Tokens 55, Peak mem 20.246 
Iter 6: Train loss 3.756, Learning Rate 1.000e-05, It/sec 9.472, Tokens/sec 104.190, Trained Tokens 66, Peak mem 20.246 
Iter 7: Train loss 3.517, Learning Rate 1.000e-05, It/sec 9.461, Tokens/sec 104.066, Trained Tokens 77, Peak mem 20.246 
Iter 8: Train loss 3.636, Learning Rate 1.000e-05, It/sec 9.443, Tokens/sec 103.877, Trained Tokens 88, Peak mem 20.246 
Iter 9: Train loss 3.653, Learning Rate 1.000e-05, It/sec 9.438, Tokens/sec 103.817, Trained Tokens 99, Peak mem 20.246                                                                  
Calculating loss...: 100%|███████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 10.85it/s]
Iter 10: Val loss 3.466, Val took 0.094s
Iter 10: Train loss 3.716, Learning Rate 1.000e-05, It/sec 9.414, Tokens/sec 103.551, Trained Tokens 110, Peak mem 20.246 GB
Saved final weights to /Users/gokdenizgulmez/Desktop/adapters/adapters.safetensors.

**Comment by Goekdeniz-Guelmez** [comment]
@awni is also mergable!

**Comment by Goekdeniz-Guelmez** [comment]
@awni I saw the changes you did the the batching PR #438 should I add the same changes into my ongoing model PRs too? This one, #392 and #429
